extensions/cl_ext_cxx_for_opencl.asciidoc

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_ext_cxx_for_opencl
+:R: pass:q,r[^(R)^]
+Khronos{R} OpenCL Working Group
+
+== Name Strings
+
+`cl_ext_cxx_for_opencl`
+
+== Contact
+
+Please see the *Issues* list in the Khronos *OpenCL-Docs* repository: +
+https://github.com/KhronosGroup/OpenCL-Docs
+
+== Contributors
+
+Kevin Petit, Arm Ltd. +
+Sven Van Haastregt, Arm Ltd. +
+Anastasia Stulova, Arm Ltd. +
+Marco Antognini, Arm Ltd. +
+Neil Hickey, Arm Ltd. +
+
+== Notice
+
+include::../copyrights.txt[]
+
+== Status
+
+Draft
+
+== Version
+
+Built On: {docdate} +
+Version: 0.1.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 3.0.2.
+
+This extension requires OpenCL 3.0 or OpenCL 2.x and `cl_khr_extended_versioning`.
```

Comment:
Are we actually using anything from `cl_khr_extended_versioning` for this extension? Maybe it's better to say that functionality here is related to `cl_khr_extended_versioning` or expected to follow similar principles?

If it's possible to avoid an extra dependency it would be better.

Reply:
`cl_version` comes from `cl_khr_extended_versioning` (well, technically that is `cl_version_khr`, but same thing).

Reply:
The `cl_version_khr` type was introduced by `cl_khr_extended_versioning`. That's the reason for the dependency.